non-nullable inputs should be marked as required

Zaid-Ajaj opened the pull request:
Fixes #335

Zaid-Ajaj commented:
Looks like lots of inputs from modules we test are now flagging a ton of their inputs as non-nullable

corymhall commented on pkg/modprovider/tfmodules.go:
```diff
@@ -365,7 +365,7 @@ func inferModuleSchema(
 		}
 
-		if variable.Default.IsNull() && !variable.Nullable {
+		if variable.Default.IsNull() && !(variable.NullableSet && variable.Nullable) {
```
> Looks like lots of inputs from modules we test are now flagging a ton of their inputs as non-nullable

Do you know how the `Nullable` and `NullableSet` gets determined? This is one of the properties from the test failures that is being read as `required` now.

https://github.com/terraform-aws-modules/terraform-aws-s3-bucket/blob/0d781fbb515b1f64c46bfba04153bef091d50fc5/variables.tf#L133-L137

Zaid-Ajaj replied:
I believe `NullableSet` means that the source code has an attribute `nullable = <true | false>` and then `Nullable` if the `nullable` is set to true. I will look again 😅

Zaid-Ajaj commented:
Auto-merged was enabled yesterday so this got merged as soon as the tests passed. 

cc @corymhall I fixed the issue using the following logic:
```go
nullable := variable.NullableSet && variable.Nullable
hasDefault := variable.Default.Type() != cty.NilType
optional := hasDefault || nullable
if !optional {
	inferredModuleSchema.RequiredInputs = append(inferredModuleSchema.RequiredInputs, key)
}
```
any default (null or otherwise) set will mean that the property is optional. Added a local test module to account for the different cases we have and the other tests passed without any changes
